src/Juvix/Prelude/Prepath.hs

```diff
@@ -12,17 +12,74 @@ where
 
 import Data.Yaml
 import Juvix.Prelude.Base
+-- import Text.Megaparsec as P
+-- import Text.Megaparsec.Char qualified as P
+-- import Text.Megaparsec.Char.Lexer qualified as P
+import Juvix.Prelude.Parsing as P
 import Juvix.Prelude.Path
 import Juvix.Prelude.Pretty
-import Juvix.Prelude.Shell
+import System.Directory (getHomeDirectory)
 import System.Directory qualified as System
+import System.Environment
 
 -- | A file/directory path that may contain environmental variables
 newtype Prepath d = Prepath {_prepath :: String}
   deriving stock (Show, Eq, Data, Generic)
 
 makeLenses ''Prepath
 
+type PrepathParts = NonEmpty PrepathPart
+
+data PrepathPart
+  = PrepathVar String
+  | PrepathString String
+  | PrepathHome
+
+-- | Paths have the following restrictions:
+-- 1. Environment variables are given with the makefile-like syntax $(VAR)
+-- 2. The three characters $ ( ) are reserved for the environment variables syntax.
+--    They cannot be part of the path.
+-- 3. ~ is reserved for $(HOME). I.e. the prepath ~~ will expand to $HOME$HOME.
+-- 4. Nested environment variables are not allowed.
+expandPrepath :: Prepath a -> IO FilePath
+expandPrepath (Prepath p) =
+  let e = parseHelper prepathParts p
+   in case e of
+        Left er -> error er
+        Right r -> expandParts r
+  where
+    prepathParts :: forall e m. MonadParsec e String m => m PrepathParts
+    prepathParts = some1 prepathPart
+      where
+        prepathPart :: m PrepathPart
+        prepathPart =
+          PrepathVar <$> evar
+            <|> PrepathString <$> str
```

Comment:
Don't you need an alternative to parse a `PrepathHome` here?

Reply:
Indeed. I'll add a test for that

Reply:
done